atom/browser/native_window_views_win.cc

```diff
@@ -299,6 +300,68 @@ bool NativeWindowViews::PreHandleMSG(UINT message,
   }
 }
 
+void NativeWindowViews::HandleSizingEvent(WPARAM w_param, LPARAM l_param) {
+  double aspect_ratio = GetAspectRatio();
+  // aspect_ratio width/height
+  if (aspect_ratio == 0) {
+    return;
+  }
+
+  RECT* window_rect;
+  window_rect = reinterpret_cast<RECT*>(l_param);
+  int width = window_rect->right - window_rect->left;
+  int height = window_rect->bottom - window_rect->top;
+
+  double result_width = 0;
+  double result_height = 0;
+
+  double temp_width = 0;
+  double temp_height = 0;
+
+  gfx::Size ratio_fit_size(width, height);
+  gfx::Size min_fit_size = GetMinimumSize();
+
+  if (min_fit_size.width() >= width) {
+    width = min_fit_size.width();
+  } else if (min_fit_size.height() >= height) {
+    height = min_fit_size.height();
+  }
+
+  switch ((UINT)w_param) {
+    case WMSZ_LEFT:
+    case WMSZ_RIGHT:
+      result_width = width;
+      result_height = static_cast<double>(width) / aspect_ratio;
```

Comment:
I don't think `static_cast<double>()` is necessary? Because `aspect_ratio` is a double, `width` should automatically be promoted.

```suggestion
      result_height = width / aspect_ratio;
```

Same goes for the next three uses of `static_cast<double>()`